pkg/cmd/root/alias.go

```diff
@@ -0,0 +1,127 @@
+package root
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/cli/cli/v2/internal/run"
+	"github.com/cli/cli/v2/pkg/findsh"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/google/shlex"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdShellAlias(io *iostreams.IOStreams, aliasName, aliasValue string) *cobra.Command {
+	return &cobra.Command{
+		Use:   aliasName,
+		Short: "Shell alias",
+		RunE: func(c *cobra.Command, args []string) error {
+			expandedArgs, err := expandShellAlias(aliasValue, args, nil)
+			if err != nil {
+				return err
+			}
+			externalCmd := exec.Command(expandedArgs[0], expandedArgs[1:]...)
+			externalCmd.Stderr = io.ErrOut
+			externalCmd.Stdout = io.Out
+			externalCmd.Stdin = io.In
+			preparedCmd := run.PrepareCmd(externalCmd)
+			if err = preparedCmd.Run(); err != nil {
+				return fmt.Errorf("failed to run external command: %w\n", err)
+			}
+			return nil
+		},
+		GroupID: "alias",
+		Annotations: map[string]string{
+			"skipAuthCheck": "true",
+		},
+		Hidden:             true,
+		DisableFlagParsing: true,
+	}
+}
+
+func NewCmdAlias(io *iostreams.IOStreams, aliasName, aliasValue string) *cobra.Command {
+	return &cobra.Command{
+		Use:   aliasName,
+		Short: fmt.Sprintf("Alias for %q", aliasValue),
+		RunE: func(c *cobra.Command, args []string) error {
+			expandedArgs, err := expandAlias(aliasValue, args)
+			if err != nil {
+				return err
+			}
+			root := c.Root()
+			root.SetArgs(expandedArgs)
+			return root.Execute()
+		},
+		GroupID: "alias",
+		Annotations: map[string]string{
+			"skipAuthCheck": "true",
+		},
+		Hidden:             true,
+		DisableFlagParsing: true,
+	}
+}
+
+// ExpandAlias processes argv to see if it should be rewritten according to a user's aliases.
+func expandAlias(expansion string, args []string) ([]string, error) {
```

Comment:
`expandAlias` and `expandShellAlias` used to be one function but they feel better and more straightforward as two. Logic in them has not changed from before though.